opentrons-ai-client/src/resources/utils/fileUtils.ts

```diff
@@ -0,0 +1,118 @@
+export const ALLOWED_FILE_TYPES = {
+  pdf: ['.pdf'],
+  csv: ['.csv'],
+  python: ['.py'],
+} as const
+
+export type FileType = 'pdf' | 'csv' | 'python' | 'unknown'
+
+export interface FileValidationResult {
+  isValid: boolean
+  error?: string
+}
+
+export interface ProcessedFileContent {
+  content: string
+  mediaType: string
+}
+
+export const FILE_SIZE_LIMITS = {
+  pdf: 10 * 1024 * 1024, // 10MB
+  csv: 2 * 1024 * 1024, // 2MB
+  python: 1 * 1024 * 1024, // 1MB
+} as const
+
+export const MAX_FILES_PER_MESSAGE = 5
+
+export const validateFile = (file: File): FileValidationResult => {
+  const fileType = getFileType(file)
+
+  if (fileType === 'unknown') {
+    return {
+      isValid: false,
+      error:
+        'Unsupported file type. Please upload PDF, CSV, or Python (.py) files.',
+    }
+  }
+
+  const sizeLimit = FILE_SIZE_LIMITS[fileType]
+  if (file.size > sizeLimit) {
+    const sizeMB = Math.round(sizeLimit / (1024 * 1024))
+    return {
+      isValid: false,
+      error: `File size too large. ${fileType.toUpperCase()} files must be under ${sizeMB}MB.`,
+    }
+  }
+
+  return { isValid: true }
+}
+
+export const getFileType = (file: File): FileType => {
+  const extension = '.' + file.name.split('.').pop()?.toLowerCase()
+
+  if (ALLOWED_FILE_TYPES.pdf.includes(extension)) return 'pdf'
+  if (ALLOWED_FILE_TYPES.csv.includes(extension)) return 'csv'
+  if (ALLOWED_FILE_TYPES.python.includes(extension)) return 'python'
```

Comment:
what you have is totally fine but we prefer to put the thing after an if statement in brackets. its easier to read and can catch bugs down the future if the return becomes more complicated.

```suggestion
  if (ALLOWED_FILE_TYPES.pdf.includes(extension)) {
  return 'pdf'
  }
  if (ALLOWED_FILE_TYPES.csv.includes(extension)) {
  return 'csv'
  }
  if (ALLOWED_FILE_TYPES.python.includes(extension)) {
  return 'python'
  }
```

Reply:
Done